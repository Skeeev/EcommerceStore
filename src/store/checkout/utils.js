import get from 'lodash/get';
import find from 'lodash/find';

const normalizeProducts = products =>
  products.map(({ key, name: title, price, total, quantity, thumb }) => ({
    id: `${key}`,
    title,
    price,
    total,
    quantity,
    imageUri: `http:${thumb}`
  }));

export const extractCheckoutData = ({ products, totals }) => ({
  products: normalizeProducts(products),
  total: get(find(totals, { id: 'total' }), 'text', '$00.00'),
  totalQuantity: products.reduce((total, { quantity }) => total + quantity, 0)
});
