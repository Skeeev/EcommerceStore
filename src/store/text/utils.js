export const prepareTextPayload = text => ({
  payload: {
    text
  }
});
