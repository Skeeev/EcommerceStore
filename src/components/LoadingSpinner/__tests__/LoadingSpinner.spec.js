import React from 'react';
import { shallow } from 'enzyme';

import { getComponentDisplayName } from 'src/utils/displayName';
import { LoadingSpinner } from '..';

describe(`${getComponentDisplayName(LoadingSpinner)} component`, () => {
  const renderComponent = props => shallow(<LoadingSpinner {...props} />);

  describe('renders correctly with', () => {
    it('default props', () => {
      const component = renderComponent();

      expect(component).toMatchSnapshot();
    });

    it('with passed props', () => {
      const component = renderComponent({
        size: 10,
        color: '#000'
      });

      expect(component).toMatchSnapshot();
    });
  });
});
