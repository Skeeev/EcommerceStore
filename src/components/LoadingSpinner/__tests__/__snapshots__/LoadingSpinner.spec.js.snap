// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner component renders correctly with default props 1`] = `
<ActivityIndicator
  animating={true}
  color="#318cbf"
  hidesWhenStopped={true}
  size={60}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;

exports[`LoadingSpinner component renders correctly with with passed props 1`] = `
<ActivityIndicator
  animating={true}
  color="#000"
  hidesWhenStopped={true}
  size={10}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
